src/pages/library/RequestDatasetAccessModal.js

```diff
@@ -92,31 +33,96 @@ const RequestDatasetAccessButton = ({ title, id }) => {
+            onDismiss()
+          }, style: { padding: '20px 30px' }
+        }, 'OK')
       ])
     ]),
-    h(ButtonPrimary, {
-      disabled: requesting || requested,
-      onClick: () => {
-        const requestAccess = _.flow(
-          Utils.withBusyState(setRequesting),
-          withErrorReporting('Error requesting dataset access')
-        )(async () => {
-          await Ajax(signal).DataRepo.requestAccess(id)
-          setRequested(true)
-        })
-        requestAccess()
+    !showWipModal && h(Modal, {
+      title: 'Request Access',
+      width: '40rem',
+      showButtons: false,
+      showX: true,
+      onDismiss
+    }, [
+      div([
+        'You cannot access this dataset because it is protected by an Authorization Domain. ',
+        'You need to obtain permission from the owner(s) of this dataset in order to get access. ',
+        'Clicking the "Request Access" button below will send an email to the admins of that dataset.'
+      ]),
+      h(IdContainer, [id => div([
+        h(FormLabel, { htmlFor: id }, ['Please tell us why']),
+        h(TextArea, {
+          id,
+          style: { height: 100, marginBottom: '0.5rem' },
+          placeholder: 'Enter your reason',
+          value: reason,
+          onChange: setReason
+        }),
+        sendCopyEnabled && div([
+          h(LabeledCheckbox, {
+            checked: sendCopy,
+            onChange: setSendCopy
+          }, [
+            label({ style: { margin: '0 2rem 0 0.25rem' } }, [`Also send me a copy`])
+          ])
+        ])
+      ])]),
+      table({ style: { margin: '1rem', width: '100%' } }, [
+        thead([
+          tr({ style: { height: '2rem' } }, [th({ style: { textAlign: 'left' } }, ['Datasets']), th({ style: { textAlign: 'left', width: '15rem' } }, ['Access'])])
+        ]),
+        tbody(
+          _.map(({ 'dct:title': title, access, 'dct:identifier': id, contacts }) => tr({ key: id, style: { height: '2rem' } }, [
+            td({ style: { paddingRight: 20 } }, [
+              title,
+              div({ style: { fontSize: '.7rem', marginTop: 5, width: 'fit-content' } }, [
+                _.map(({ email }) => email && h(Link, { key: email, href: `mailto:${email}`, style: { marginTop: 5, display: 'block' } }, [email]), contacts)
+              ])
+            ]),
+            td([
+              Utils.switchCase(access,
+                ['Controlled', () => h(RequestDatasetAccessButton, { title, id, setShowWipModal })],
+                ['Pending', () => span({ style: { fontWeight: 600 } }, ['Request Pending'])],
+                [Utils.DEFAULT, () => span({ style: { fontWeight: 600 } }, ['Permission Granted'])]
+              )
+            ])
+          ]), datasets)
+        )
+      ])
+    ])
+  ])
+}
+
+const RequestDatasetAccessButton = ({ title, id, setShowWipModal }) => {
+  const [requesting, setRequesting] = useState(false)
+  const [requested, setRequested] = useState(false)
```

Comment:
How would you feel about either merging these into one state, where it switches the value. The values can be stored in an enum. Having two mutually exclusive states opens the possibility for the code to get into what should be an impossible state. Or we could remove the `requesting` state for now, it looks like it is not used.

Reply:
That's a great point! Let me take a look